src/wkt/src/timestamp.rs

```diff
@@ -271,6 +271,9 @@ impl TryFrom<&str> for Timestamp {
         let nanos_since_epoch = odt.unix_timestamp_nanos();
         let seconds = (nanos_since_epoch / NS) as i64;
         let nanos = (nanos_since_epoch % NS) as i32;
+        if !(0..Self::NS).contains(&nanos) {
+            return Timestamp::new(seconds - 1, Self::NS + nanos);
```

Comment:
nit: `Self::NS` is just called `NS` above.

Reply:
Oh, they are different types. Ignore me.

Reply:
I got confused myself.